src/ast.ml

```diff
@@ -288,6 +288,9 @@ and
   | RealLit of loc * num * float_literal_suffix option
   | StringLit of loc * string (* string literal *)
   | ClassLit of loc * string (* class literal in java *)
+  | CxxTypeid of loc * expr (* typeid(expr) in C++ *)
+  | WCxxTypeid of loc * expr * type_ (* typechecked typeid(expr) *)
+  | CxxTypeInfo of loc * type_ (* TypeInfo object in C++ *)

```

Comment:
As mentioned, I will be using the type info stuff for C programs as well, to check effective types.

Reply:
Removed the `Cxx` prefix from `CxxTypeInfo`